### ISSUE TITLE
BUG: Scipy installs and uses 2 copies of libopenblas (indirectly)

### ISSUE THREAD

**Issue by rjungbeck** [scipy/scipy#17242]
### Describe your issue.

Scipy requires numpy. 

But both include (at least in their Windows wheels) a form of libopenblas.

So when you want to redistribute an app that is using scipy you are forced to redistribute 2 copies of libopenblas  (each 35 MB) that both get loaded into memory.

A simple solution would be to publish libopenblas as an own library on Pypi  (and make scipy and numpy depending on it).
An alternative solution would be for scipy to use the numpy installed libopenblas.

### Reproducing Code Example

```python
pip install scipy
```

### Error message

```shell
No error message
```

### SciPy/NumPy/Python version information

1.91/1.23.4/3.10.8

**Comment by ilayn** [comment]
It is unfortunately not that simple. For two major and lots of minor reasons; numpy and scipy openblas versions can be and typically **are** different. In the meantime, say scipy can be using MKL while numpy has still openblas and all the combinations there of. So the openblas they are shipping with are not necessarily the same and unfortunately differences between versions of openblas are quite non-negligible. 

If we tie up to numpy's openblas then there will be an exceedingly tight coupling between two packages and that is very difficult to maintain especially when scipy goes one version up but numpy arrives two months later having an oscillation of releases instead of simultaneity. That is something next to impossible; imagine my NumPy version is 1.21.4 and scipy is 1.9.2. which openblas should we use ?

Note that while BLAS/LAPACK version changes very slowly but OpenBLAS releases much more often (because it does more stuff) hence it is not the same thing.

**Comment by rgommers** [comment]
> A simple solution would be to publish libopenblas as an own library on Pypi (and make scipy and numpy depending on it).

This is being worked on - see gh-15129 and https://github.com/MacPython/openblas-libs/issues/86.

There are nontrivial issues to think about though, as @ilayn points out. I'd say this is a duplicate of gh-15129, so I propose to close this issue.

**Comment by rjungbeck** [comment]
If you understand gh-15129 to not only cover the Mac performance issue described in it,  but the root cause, it is.

I guess I have the same problem that many people who want to deliver PyData based apps have: We get extremely large installer and installation sizes (even for simple apps). And the biggest library (25% / 35% of the SciPy and 66% / 76% of the NumPy wheel / installed size) is libopenblas. So I think a solution could benefit many.

**Comment by rgommers** [comment]
Yes, wheel file size is another motivation. We already managed to get rid of ~10 MB in the `manylinux` wheels between `1.9.1` and `1.9.2`. Removing data/test files is another large win for all wheels. And one OpenBLAS copy is too. Plus combining some Cython extensions (Cython is quite greedy). I think we can get it down to below 20 MB for SciPy.

I did mention wheel size in gh-15129, but it'd be good to make it more explicit. Please feel free to add your comment there.
